Core/include/Acts/Navigation/MultiWireLayerUpdators.hpp

```diff
@@ -0,0 +1,199 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Definitions/Common.hpp"
+#include "Acts/Detector/DetectorVolume.hpp"
+#include "Acts/Detector/Portal.hpp"
+#include "Acts/Detector/detail/GridAxisGenerators.hpp"
+#include "Acts/Detector/detail/IndexedGridFiller.hpp"
+#include "Acts/Detector/detail/IndexedSurfacesGenerator.hpp"
+#include "Acts/Detector/detail/ReferenceGenerators.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Navigation/NavigationState.hpp"
+#include "Acts/Navigation/NavigationStateFillers.hpp"
+#include "Acts/Navigation/NavigationStateUpdators.hpp"
+#include "Acts/Navigation/SurfaceCandidatesUpdators.hpp"
+#include "Acts/Surfaces/Surface.hpp"
+#include "Acts/Utilities/detail/Axis.hpp"
+#include "Acts/Utilities/detail/Grid.hpp"
+
+#include <memory>
+#include <tuple>
+
+namespace Acts {
+namespace Experimental {
+
+using grid_type = detail::GridAxisGenerators::EqBoundEqBound::grid_type<
+    std::vector<std::size_t>>;
+using IndexedSurfacesGridType = IndexedSurfacesImpl<grid_type>;
+
+struct MultiWireLayerImpl : public INavigationDelegate {
+  IndexedSurfacesGridType getIndexedSurfacesFromVolume(
+      NavigationState& nState) const {
+    auto gridSurfaces = nState.currentVolume->surfaces();
+    GeometryContext gctx;
+
+    std::array<std::pair<float, float>, 3> min_max_multilayer;
+    std::fill(min_max_multilayer.begin(), min_max_multilayer.end(),
+              std::make_pair<float, float>(std::numeric_limits<float>::max(),
+                                           -std::numeric_limits<float>::max()));
+
+    for (auto& gridSurf : gridSurfaces) {
+      min_max_multilayer[0].first = std::min(min_max_multilayer[0].first,
+                                             (float)gridSurf->center(gctx).x());
+      min_max_multilayer[0].second = std::max(
+          min_max_multilayer[0].second, (float)gridSurf->center(gctx).x());
+
+      min_max_multilayer[1].first = std::min(min_max_multilayer[1].first,
+                                             (float)gridSurf->center(gctx).y());
+      min_max_multilayer[1].second = std::max(
+          min_max_multilayer[1].second, (float)gridSurf->center(gctx).y());
+
+      min_max_multilayer[2].first = std::min(min_max_multilayer[2].first,
+                                             (float)gridSurf->center(gctx).z());
+      min_max_multilayer[2].second = std::max(
+          min_max_multilayer[2].second, (float)gridSurf->center(gctx).z());
+    }
+    auto radius = gridSurfaces.front()->bounds().values()[0];
+
+    float binWidthX = 2 * radius, binWidthY = sqrt(3) * radius;
```

Comment:
Can we make these values being configurable? I have hard time justifying `sqrt(3)` 
Maybe the default value can be the number width / number of surfaces in the two directions?